src/builtin.jq

```diff
@@ -264,6 +264,11 @@ def pick(pathexps):
 # ensure the output of debug(m1,m2) is kept together:
 def debug(msgs): (msgs | debug | empty), .;
 
+# assert property with message
+# success passes the original input through
+# failure raises an error with the original error message
+def assert(test; $message): if test then . else error("assertion error: \($message)") end;
```

Comment:
If this drops the `$` on `$message`, it'll only evaluate the expression if there's an error, right?  That doesn't matter for simple messages, but might be relevant for more complex ones.

```suggestion
def assert(test; message): if test then . else error("assertion error: \(message)") end;
```